src/Game.qml

```diff
@@ -0,0 +1,609 @@
+                                    break;
+                            } else {
+                                if (!moveCell(x, j, x, y))
+                                    break;
+                            }
+                        }
+                    }
+                }
+            }
+
+            if (gesture == "right" || gesture == "down") {
+                for (var x=0; x<rows; x++) {
+                    for (var y=cols-1; y>=0; y--) {
+                        for (var j= y-1; j>=0; j--) {
+                            if (gesture == "right") {
+                                if (!moveCell(j, x, y, x))
+                                    break;
+                            } else {
+                                if (!moveCell(x, j, x, y))
+                                    break;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        function synchronize() {
+            gridScore.value = score
+            var cellString = ""
+            for (var x=0;x<rows;x++) {
+                for (var y=0; y<cols; y++) {
+                    cellString = cellString + cells[x][y] + ","
+                }
+                cellString = cellString + ";"
+            }
+            gridCells.value = cellString
+        }
+
+        function restore() {
+            if (gridCells.value.length == 0) return false
+
+            score = gridScore.value
+            for (var i=0; i<rows*cols;i++) {
+                emptyGridCells[i] = i
+                cellsGrid.itemAt(i).x1 = -1
+                cellsGrid.itemAt(i).y1 = -1
+                cellsGrid.itemAt(i).val = 0
+            }
+
+            var emptyCells = 0
+            cells = gridCells.value.split(";")
+            for (var x=0;x<rows;x++) {
+                cells[x] = cells[x].split(",")
+                for (var y=0;y<cols;y++) {
+                    cells[x][y] = parseInt(cells[x][y])
+
+                    // UI
+                    if (cells[x][y] != 0) {
+                        var emptyGrid = emptyGridCells[0]
+                        emptyGridCells.shift()
+
+                        cellsGrid.itemAt(emptyGrid).animateMove = false
+                        cellsGrid.itemAt(emptyGrid).val = cells[x][y]
+                        cellsGrid.itemAt(emptyGrid).x1 = x
+                        cellsGrid.itemAt(emptyGrid).y1 = y
+                        cellsGrid.itemAt(emptyGrid).animateMove = true
+                    } else {
+                        emptyCells ++
+                    }
+                }
+            }
+
+            if ((emptyCells === 0) && !movePossible()) {
+                console.log("restore::Moving is no longer possbile! GAME OVER!!")
+                bestScore = Math.max(bestScore, score)
+                gameOver.visible = true
+                gameOver.angle = -90
+            }
+            return true
+        }
+
+
```

Comment:
Unnecessary empty line.
Instead of removing it, I would replace it with a comment about the purpose of `randCell()`